kafka/producer.c

```diff
@@ -334,17 +334,16 @@ lua_producer_produce(struct lua_State *L) {
     lua_gettable(L, -2 );
     // rd_kafka will copy key so no need to worry about this cast
     char *key = (char *)lua_tostring(L, -1);
```

Comment:
But why didn't you simply use `lua_tolstring` that accepts an additional argument where string will be returned.